In [8]:
import requests

### Q1: dlt version

In [1]:
!pip install -q "dlt[qdrant]" "qdrant-client[fastembed]"

In [ ]:
import dlt
print('Q1 - answer:', dlt.__version__)

'1.13.0'

### dlt Resource

In [3]:
@dlt.resource
def zoomcamp_data():
    docs_url = 'https://github.com/alexeygrigorev/llm-rag-workshop/raw/main/notebooks/documents.json'
    docs_response = requests.get(docs_url)
    documents_raw = docs_response.json()

    for course in documents_raw:
        course_name = course['course']

        for doc in course['documents']:
            doc['course'] = course_name
            yield doc

### Q2 

In [4]:
from dlt.destinations import qdrant

qdrant_destination = qdrant(
  qd_path="db.qdrant", 
)

In [9]:
pipeline = dlt.pipeline(
    pipeline_name="zoomcamp_pipeline",
    destination=qdrant_destination,
    dataset_name="zoomcamp_tagged_data"

)
load_info = pipeline.run(zoomcamp_data())
print(pipeline.last_trace)

Run started at 2025-07-10 19:49:12.647497+00:00 and COMPLETED in 7.68 seconds with 4 steps.
Step extract COMPLETED in 0.65 seconds.

Load package 1752176953.2579715 is EXTRACTED and NOT YET LOADED to the destination and contains no failed jobs

Step normalize COMPLETED in 0.21 seconds.
Normalized data for the following tables:
- zoomcamp_data: 948 row(s)
- _dlt_pipeline_state: 1 row(s)

Load package 1752176953.2579715 is NORMALIZED and NOT YET LOADED to the destination and contains no failed jobs

Step load COMPLETED in 6.22 seconds.
Pipeline zoomcamp_pipeline load step completed in 6.21 seconds
1 load package(s) were loaded to destination qdrant and into dataset zoomcamp_tagged_data
The qdrant destination used c:\Users\ROG\Desktop\Sandra\repos_sancyb\llm-zoomcamp-sancyb\homeworks\db.qdrant location to store data
Load package 1752176953.2579715 is LOADED and contains no failed jobs

Step run COMPLETED in 7.68 seconds.
Pipeline zoomcamp_pipeline load step completed in 6.21 seconds
1 loa

**Q2 - answer: 948 row(s)**

### Q3 - Embeddings

**Q3 answer - "fast-bge-small-en"**